scl/logscale/logscale.conf

```diff
@@ -0,0 +1,76 @@
+#############################################################################
+# Copyright (c) 2023 Attila Szakacs
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 2 as published
+# by the Free Software Foundation, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# As an additional exemption you are allowed to compile & link against the
+# OpenSSL libraries as published by the OpenSSL project. See the file
+# COPYING for details.
+#
+#############################################################################
+
+# Batching best practices: https://library.humio.com/falcon-logscale/api-ingest.html#api-ingest-best-practices
+
+@requires json-plugin
+
+
+# https://library.humio.com/falcon-logscale/api-ingest.html#api-ingest-structured-data
+block destination logscale(
+  url("https://cloud.humio.com")
+  token()
+
+  rawstring("${MESSAGE}")
+  timestamp("${S_ISODATE}")
+  timezone("")
+  attributes("--scope rfc5424 --exclude MESSAGE --exclude DATE --leave-initial-dot")
+
+  batch_lines(5000)
```

Comment:
Thanks!

I have changed `batch_lines()`, `batch_bytes()` and `batch_timeout()`.

Is there any particular reason for setting `workers(20)`? In the source code there is a hard limit on the maximum number of workers, and we have multiple `http()` SCLs and their workers add up. We are trying to balance a bit, so even if a user configures multiple SCL based drivers, they won't hit the hard limit. Because of this I would keep it 8, if you don't mind.

Reply:
if the response time from the http server is 500ms (round trip) we would get about 2 posts per second per thread limiting the throughput to around 20Mbs or 20k eps if we had 10 workers I was going for double that as a modest size firewall source alone can run around that level. http posts are mostly network wait state so a higher number of works is really needed to keep throughput up

Reply:
These are good points. I think that we should probably separate the concept of a worker and a batch within our HTTP destination. We might be building batches, some of which are being transferred to the server, while the others are still waiting for more data. This way threads would only be occupied while the HTTP transaction happens, thus the number of threads may be a lot lower than the number of batches being built.

If the total number of threads exceeds the compile time constant of 256, we will get this message:

      msg_warning_once("Unable to allocate a unique thread ID. This can only "
                       "happen if the number of syslog-ng worker threads exceeds the "
                       "compile time constant MAIN_LOOP_MAX_WORKER_THREADS. "
                       "This is not a fatal problem but can be a cause for "
                       "decreased performance. Increase this number and recompile "
                       "or contact the syslog-ng authors",
                       evt_tag_int("max-worker-threads-hard-limit", MAIN_LOOP_MAX_WORKER_THREADS));

This only causes decreased performance in the source related threads case, so we might just get rid off this problem by not allocating a thread id for HTTP worker threads, which will never use the thread id at the moment.

Reply:
Merged it with `workers(8)`, as I would like to generate a container image from master and this PR already had an approve, but opened a PR to raise the workers to 20: #4488